src/server/game/Battlegrounds/BattlegroundQueue.cpp

```diff
@@ -957,6 +961,7 @@ void BattlegroundQueue::BattlegroundQueueUpdate(uint32 /*diff*/, BattlegroundTyp
                 if (!(*itr3)->IsInvitedToBGInstanceGUID
                     && (((*itr3)->ArenaMatchmakerRating >= arenaMinRating && (*itr3)->ArenaMatchmakerRating <= arenaMaxRating)
                         || (int32)(*itr3)->JoinTime < discardTime)
+                    && ((*itr_teams[0])->ArenaTeamId != (*itr3)->PreviousOpponentsTeamId || ((int32)(*itr3)->JoinTime < discardOpponentsTime))
```

Comment:
I realize this is a copypaste of rating discarding but this doesn't behave like your quote describes - what this does is prevent queueing into the same team for the first 2 minutes of being in queue, not a 2 minute addition to average wait time

Reply:
@Gooyeth

Reply:
@Gooyeth

Reply:
> what this does is prevent queueing into the same team for the first 2 minutes of being in queue

This seems like the correct behavior, the blue post says 

> There is currently a 2-minute queue time addition to play against a team you just played against (so, 2 minutes plus normal queue time to be matched against them).

Reply:
No, example:
average wait time: 5 minutes
intended behavior: waiting for ~7 minutes before seeing the same team
current code: restricted only for 2 minutes from the moment of joining queue

Reply:
Isn't "average wait time" only a calculated value based on how much time the previous ten teams spent in queue?

I mean, if one team spends five minutes in queue before another team joins, then they'll contribute five minutes to the "average wait time".

Once two teams are in queue and allowed to match up, the arena match starts. If both join at the same time, their wait time is basically zero.

We forbid the same two teams from facing each other for the first two minutes in queue, but since they're both in queue, their wait time should be two minutes (arenas start when the system finds two available teams,

Unless I'm missing some big piece in the picture, in that case I apologize :P